libs/execution_context/ebpf_core.c

```diff
@@ -491,34 +503,32 @@ _ebpf_core_protocol_map_get_next_key(
 {
     ebpf_result_t retval;
     ebpf_map_t* map = NULL;
-    const uint8_t* previous_key;
-    uint8_t* next_key = NULL;
+    size_t previous_key_length;
+    size_t next_key_length;
 
     retval = ebpf_reference_object_by_handle(request->handle, EBPF_OBJECT_MAP, (ebpf_object_t**)&map);
     if (retval != EBPF_SUCCESS)
         goto Done;
 
-    const ebpf_map_definition_t* map_definition = ebpf_map_get_definition(map);
+    retval = ebpf_safe_size_t_subtract(
+        request->header.length,
+        EBPF_OFFSET_OF(ebpf_operation_map_get_next_key_request_t, previous_key),
+        &previous_key_length);
```

Comment:
key length cannot be variable. It can safely be read from map_definition IMO.

Reply:
This is the size of key supplied by the untrusted user program.

Reply:
Agreed.